doc/nrf/ug_dev_model.rst

```diff
@@ -27,6 +27,23 @@ Basic familiarity with Git is required to understand the architecture of the rep
 
 All |NCS| repositories are publicly hosted on `GitHub`_, and accessible to both individual users and companies.
 
+A note about forks, upstreams and downstreams
+=============================================
+
+Some key concepts used throughout this guide deserve their own introduction to avoid confusion among users not experienced with Git and multiple repositories.
+
+Git is a distributed version control system that allows repositories to be easily duplicated, creating copies of them. Every time you take an existing Git repository and create a copy of it, you are effectively creating a *fork* of that repository. This means that you now have an identical copy that might diverge from the original over time, since commits to the original will not be automatically reflected in the copy.
+
+.. note:
+   When we talk about forks or copying Git repositories we refer to the creation
+   of a new repository hosted in a server and accessible to other users. If you
+   are simply cloning a repository to your local machine then that is referred
+   to as a *clone*, and not a fork.
+
+Whenever you create a fork by creating a copy of an existing repository, the original repository can be called the *upstream*, and the newly created copy conversely the *downstream*.
+
+A fork can be hosted in any server, including public Git hosting site such as `GitHub`_. It is however important to differentiate between the generic concept of fork introduced earlier in this section and GitHub's own concept of a `GitHub fork`_.  A GitHub fork is GitHub's own concept of a fork: it includes both creating a copy of the original repo being forked and also tagging the downstream or fork with a flag that allows users to send Pull Requests from the fork to its upstream. GitHub also supports creating forks without using this particular feature, simply by creating an empty repository and then pushing the contents of the repository to be cloned to it manually.
```

Comment:
s/public/a public/

I am not sure pull request is a proper noun phrase in this context. Should be lowercase if not.

Are you sure about the correctness of "GitHub also supports creating forks without using this particular feature, simply by creating an empty repository and then pushing the contents of the repository to be cloned to it manually."?

If you make an empty repository, there is no fork relationship in the GitHub sense AFAIK. Is there some way to manually say "I'm a fork of this other repo over here"? If so, might want to say so in this doc, which only mentions pushing.

Reply:
GitHub allows forks in the common sense of the work, it's documented here:
https://help.github.com/en/articles/duplicating-a-repository
I will rework this section to reflect that.